torch/csrc/autograd/variable.cpp

```diff
@@ -28,7 +28,10 @@ AutogradMeta::AutogradMeta(at::TensorImpl* self_impl, bool requires_grad, Edge g
   output_nr_ = gradient_edge.input_nr;
 
   // set_requires_grad also checks error conditions.
-  set_requires_grad(requires_grad, self_impl);
+  if (requires_grad) {
+    TORCH_INTERNAL_ASSERT(self_impl);
+    set_requires_grad(requires_grad, self_impl);
```

Comment:
Do we assume that `set_requires_grad` is always no-op when `requires_grad` is false? (Not sure if this is the semantics we are moving to)

Reply:
In general, `set_requires_grad` is not a no-op when `requires_grad`, but in this particular case, as the factory function for AutogradMeta, it is indeed a no-op, because the default setting for `requires_grad` false.

Reply:
When a Variable has AutogradMeta and the user calls `set_requires_grad(false)`, do we set the AutogradMeta to nullptr again? (Is this logic handled elsewhere?)

Reply:
We discussed this offline, but as this patch stands right now, the answer is no (because you have to do a consistency check on the rest of AutogradMeta to see if it matches the default constructed form). I agreed to add a comment about this.